control_panel_api/models.py

```diff
@@ -24,6 +24,16 @@ def get_full_name(self):
     def get_short_name(self):
         return self.name
 
+    @property
+    def aws_role_name(self):
+        return f"{settings.ENV}_user_{self.username.lower()}"
+
+    def aws_create_role(self):
+        services.create_user_role(self.aws_role_name)
```

Comment:
Why this is called `create_user_role()` but not `create_role()`?
I would imagine because create the IAM role for a user and create the IAM role for the app involves slightly different things.

My point would be, I wonder if we should also keep `services.delete_user_role()`/`services.delete_app_role()`. I see why you renamed it to `services.delete_role()` to remove the duplication, still one part of me thinks they're conceptually two different things.

BUT, this is just food for thought, if you prefer to have `services.delete_role()` for the time being, I wouldn't mind. I'll have a look at the rest in case I see any reasons to not go this way.

Reply:
Because as per the method you created `create_app_role` this method encapsulates the policy document.

Reply:
@xoen yes I'm going to merge them in the correct order.